src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -155,6 +180,17 @@ public void visitToken(DetailAST ast)
     }
 
     /**
+     * Determines whether enhanced for-loop variable can be checked or not.
+     * @param ast The ast to compare.
+     * @return true if enhanced for-loop variable can be checked.
+     */
+    private boolean canCheckEnhancedForLoopVariable(DetailAST ast)
```

Comment:
I think we always can check this. The question is if we should :) Rename method and Javadoc to `shouldCheckEnhancedForLoopVariable`.